### DISCUSSION TITLE
ERROR in main Module not found: Error: Can't resolve './src' in

### DISCUSSION THREAD

**Discussion by Yokubjon-J** [webpack/webpack#16028] [Q&A] [answered]
Hello! This is the first time I am using webpack. I decided to use it instead of create-react-app, but right now I am facing what seems to be a very difficult problem with webpack: I am getting the following error:
```
ERROR in main
Module not found: Error: Can't resolve './src' in 'C:\Users\yakubdjon.juraev\Desktop\p\chat-app'
resolve './src' in 'C:\Users\yakubdjon.juraev\Desktop\p\chat-app'
  using description file: C:\Users\yakubdjon.juraev\Desktop\p\chat-app\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: C:\Users\yakubdjon.juraev\Desktop\p\chat-app\package.json (relative path: ./src)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
```
And here is the screenshot:
![image](https://user-images.githubusercontent.com/67240021/177996081-da3bfb07-36f7-4b83-9cbb-01a256985f2f.png)
I find this error peculiar because I don't have any `src` folder in my project.
I tried to identify the error several times but it has got really hard for me to find a fix. Could you please help me?
Here is my repo: https://github.com/Yokubjon-J/chat-app (I run my app with `npm run development`).

**Comment by vankop** [comment]
instead of https://github.com/Yokubjon-J/chat-app/blob/no-cra/webpack.config.client.production.js#L29 you should `return config`.
_______

btw if function does nothing you can `module.exports = config;`

**Reply by Yokubjon-J** [reply]
Unfortunately, I am still getting an error, but this time it is slightly different:
![image](https://user-images.githubusercontent.com/67240021/178102962-63a27112-c141-43a2-a7f7-0a9ab00c5c4d.png)

**Comment by Yokubjon-J** [comment]
I guess that the error is caused because webpack is not generating `dist/bundle.js` file, which `template.js` references: https://github.com/Yokubjon-J/chat-app/blob/no-cra/template.js#L10. `template.js` is in turn served by express.

bundle.js should be generated at this line https://github.com/Yokubjon-J/chat-app/blob/no-cra/server/devBundle.js#L8 of `server/devBundle.js` file.

@vankop could you please point out why webpack is not generating `bundle.js`.

**Comment by Wuyue0** [comment, marked as answer]
经过我运行你的项目 我发现在经过compile函数的时候 获取的webpackconfig 配置是空的 进而报错。 当我修改完毕后 你的webpack.config.client文件的配置 传递给webpack函数 配置会报错 现在没有具体发现是哪一行。 但是你上边贴图的错算是解决了 。 因此对于你上边的图报错 原因就是 传递给webpack参数为{}  。

After I run your project, I find that the ‘webpackConfig’ parameter obtained when I pass the compile function is empty, and then an error is reported. When I finish modifying your webpack config. When the configuration of the client file is passed to the webpack function, an error will be reported. At present, no specific line of configuration has been found. But the mistake of your upper mapping is solved. Therefore, the reason for the error in the above figure is that the parameter passed to webpack is {}.

![10d18ea2d1742ff534007da69f37568](https://user-images.githubusercontent.com/39546369/178112888-35ad2378-e369-42da-b492-484c22d66b95.png)
